test/unit-test/mqtt_agent_utest.c

```diff
@@ -853,3 +909,194 @@ void test_MQTTAgent_Terminate_success( void )
+
+    mqttStatus = MQTTAgent_CommandLoop( &mqttAgentContext );
+
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
+}
+
+/**
+ * @brief Test MQTTAgent_CommandLoop behavior when sent command is not NULL.
+ */
+void test_MQTTAgent_CommandLoop_with_sent_command( void )
+{
+    MQTTStatus_t mqttStatus;
+    MQTTAgentContext_t mqttAgentContext;
+    MQTTAgentCommandFuncReturns_t pReturnFlags;
+    Command_t command_to_send;
+
+    setupAgentContext( &mqttAgentContext );
+    mqttAgentContext.mqttContext.connectStatus = MQTTConnected;
+    pReturnFlags.addAcknowledgment = false;
+    pReturnFlags.runProcessLoop = true;
+    pReturnFlags.endLoop = true;
+
+    MQTTAgentCommand_Publish_ExpectAnyArgsAndReturn( MQTTSuccess );
+    MQTTAgentCommand_Publish_ReturnThruPtr_pReturnFlags( &pReturnFlags );
+    MQTT_ProcessLoop_IgnoreAndReturn( MQTTSuccess );
+
+    /* Initializing command to be sent to the commandLoop. */
+    command_to_send.commandType = PUBLISH;
+    command_to_send.pCommandCompleteCallback = stubCompletionCallback;
+    command_to_send.pCmdContext = NULL;
+    command_to_send.pArgs = NULL;
+    mqttAgentContext.agentInterface.pMsgCtx->pSentCommand = &command_to_send;
+
+    mqttStatus = MQTTAgent_CommandLoop( &mqttAgentContext );
+
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
+    /* Making Sure that callback is called. */
+    TEST_ASSERT_EQUAL( 1, commandCompleteCallbackCount );
+}
+
+/**
+ * @brief Test that MQTTAgent_CommandLoop adds acknowledgment in pendingAcks for the commands requiring
+ * acknowledgments.
+ */
+void test_MQTTAgent_CommandLoop_add_acknowledgment_success( void )
+{
+    MQTTStatus_t mqttStatus;
+    MQTTAgentContext_t mqttAgentContext;
+    MQTTAgentCommandFuncReturns_t pReturnFlags;
+    Command_t command_to_send;
+
+    setupAgentContext( &mqttAgentContext );
+    mqttAgentContext.mqttContext.connectStatus = MQTTConnected;
+    pReturnFlags.addAcknowledgment = true;
+    pReturnFlags.runProcessLoop = true;
+    pReturnFlags.endLoop = true;
+    pReturnFlags.packetId = 1U;
+
+
+    MQTTAgentCommand_Publish_ExpectAnyArgsAndReturn( MQTTSuccess );
+    MQTTAgentCommand_Publish_ReturnThruPtr_pReturnFlags( &pReturnFlags );
+    MQTT_ProcessLoop_IgnoreAndReturn( MQTTSuccess );
+
+    /* Initializing command to be sent to the commandLoop. */
+    command_to_send.commandType = PUBLISH;
+    command_to_send.pCommandCompleteCallback = stubCompletionCallback;
+    command_to_send.pCmdContext = NULL;
+    command_to_send.pArgs = NULL;
+
+    mqttAgentContext.agentInterface.pMsgCtx->pSentCommand = &command_to_send;
+    mqttStatus = MQTTAgent_CommandLoop( &mqttAgentContext );
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
+
+    /* Making Sure that acknowledgement is added. */
+    TEST_ASSERT_EQUAL( 1, mqttAgentContext.pPendingAcks[ 0 ].packetId );
+    TEST_ASSERT_EQUAL_MEMORY( &command_to_send, mqttAgentContext.pPendingAcks[ 0 ].pOriginalCommand, sizeof( Command_t ) );
```

Comment:
It's not just the memory that needs to be equal, the address should also be the same
```suggestion
    TEST_ASSERT_EQUAL_PTR( &command_to_send, mqttAgentContext.pPendingAcks[ 0 ].pOriginalCommand );
```